### PR TITLE
Fluid properties

### PR DESCRIPTION
**Pull request by jb803**
We will often be using air or a normalised gas using the same value of gamma. A function that writes a standard thermophysicalProperties dict for these two gases would therefore be helpful.

This has been implemented in firefish.fluids. The supersonic wedge example has been updated to use this new functionality.

### CODE REVIEW

File: firefish/fluids.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+"""
+This module provides shortcuts to commonly used fluids
+"""
+
+import enum
+
+from firefish.case import (
+    FileName)
```
Comment (rjw57) [inline firefish/fluids.py]: Change this to

```python
from firefish.case import FileName
```

?

File: firefish/fluids.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+class Fluid(enum.Enum):
+    """
+    An enumeration of commonly used fluids
+
+    AIR generates the recommended OpenFoam thermophysicalProperties for air.
+
+    DIMENSIONLESS_AIR generates a normalised gas whith gamma=7/5 and with
+    the property that at 1 temperature unit the speed of sound is 1
+    velocity unit"""
```
Comment (rjw57) [inline firefish/fluids.py]: Tiny nitpick here: the convention for multiline docstrings is to have a blank line at the end:

```python
class Fluid(enum.Enum):
    """This is
    a multiline docstring

    """
    AIR=0

etc, etc.

File: firefish/fluids.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+def write_thermophysical_properties(case, fluid):
+    """"
+    Writes a thermophysicalProperties dict in the given case for the
+    specified fluid.
+
+    Args:
+        case (firefish.case.Case): the case in which to write the dict
+        fluid (firefish.fluids.Fluid): the fluid to use
+    """
```
Comment (rjw57) [inline firefish/fluids.py]: See above, add blank line above ``"""`` and *remove* it below.

File: firefish/fluids.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import enum
+
+from firefish.case import (
+    FileName)
+
+class Fluid(enum.Enum):
+    """
+    An enumeration of commonly used fluids
+
+    AIR generates the recommended OpenFoam thermophysicalProperties for air.
```
Comment (rjw57) [inline firefish/fluids.py]: Is there something in Google's style guide about enum-likes? I'd have expected something like

```
Attributes:
    AIR: the recommended OpenFoam thermophysicalProperties for air
    DIMENSIONLESS_AIR: a normalised gas whith gamma=7/5 and...
```

But this is based entirely on what looks surprising. I've no evidence on what is recommended. That being said, listing them as attributes will probably make the docs look nicer.

File: firefish/fluids.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    Writes a thermophysicalProperties dict in the given case for the
+    specified fluid.
+
+    Args:
+        case (firefish.case.Case): the case in which to write the dict
+        fluid (firefish.fluids.Fluid): the fluid to use
+    """
+
+    thermo_dict = None
+
```
Comment (rjw57) [inline firefish/fluids.py]: Can we get a comment with the source for these values? Probably a link to the OpenFOAM docs?

File: test/test_fluids.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
+"""
+Test writing of default thermophysical properties
+"""
+import os
+
+import pytest
+
+from firefish.case import (
+    Case)
```
Comment (rjw57) [inline test/test_fluids.py]: This should probably be

```python
from firefish.case import Case
```

**Comment by rjw57** [comment]
Is ``firefish.fluids`` the best name for the module? I can see other standard dictionaries appearing in the future so maybe something like ``firefish.materials`` or ``firefish.properties``?

**Comment by rjw57** [comment]
Addendum: I suppose one could look at the various ``write_{...}`` functions in the examples to see what other things may belong in this module. Naming things is hard :frowning:.

**Comment by jb803** [comment]
Yeah - I wasn't sure about the name either. How about something like `firefish.commondicts` ?

**Comment by rjw57** [comment]
Yes. Although it's annoying that "dict" means different things in Python and OpenFOAM. Since the functions all operate on ``Case`` objects, perhaps it is best to move them into the ``case`` module? The Pythonic idiom is to have as few modules as possible.

**Comment by jb803** [comment]
Ok - and then I guess have `write_std_thermophysical_properties()` as one of the functions inside the `Case` class?

**Comment by rjw57** [comment]
No, just add it as a function in the ``case`` module. I don't think it belongs in the ``Case`` class proper.

**Comment by rjw57** [comment]
Rationale: modules collect related functionality. Classes provide safe access to internal state.

**Comment by rjw57** [comment]
(Having "everything is a class" and "one class contains everything" is a bit of a symptom of Java Enterprise Programming I find.)

**Comment by jb803** [comment]
This should take into account all the comments above and moves it to `firefish.case`

File: README.md [hunk 7]
```diff
@@ -3,6 +3,8 @@
 [![Build Status](https://travis-ci.org/cuspaceflight/firefish.svg?branch=master)](https://travis-ci.org/cuspaceflight/firefish)
 [![Coverage
 Status](https://coveralls.io/repos/cuspaceflight/firefish/badge.svg?branch=master&service=github)](https://coveralls.io/github/cuspaceflight/firefish?branch=master)
+[![Documentation
```
Comment (rjw57) [inline README.md]: I'm not sure this should be here. Was there a ``git rebase`` snafu?

**Comment by rjw57** [comment]
:+1: for merging if tests pass.

File: README.md (hunk 7 above)
Comment (jb803) [inline README.md]: Yeah - I accidentally managed to pull Boyuan's stlJoin branch into my master whilst playing around. Wasn't sure how to undo it and so deleted everything and started again. It's probably an artifact from that I guess.

**Pull request merged**